### ISSUE TITLE
Reproducibility Issue

### ISSUE THREAD

**Issue by jeongwhanchoi** [YimiAChack/STDDE#2]
Hello @YimiAChack ,

I have a question regarding the reproducibility of your code. The paper does not provide all the best hyperparameters for each dataset. Using the settings that I could infer, I ran the PeMSD4 experiment and obtained the following results, which are significantly different from those reported in the paper:
- RMSE: 32.63
- MAE: 22.42
- MAPE: 16.33

Below is the log from my experiment. I would greatly appreciate it if you could address this issue promptly.

Thank you

[log_2024-12-24_10-54-45_514918.log](https://github.com/user-attachments/files/18238175/log_2024-12-24_10-54-45_514918.log)

**Comment by sxgty** [comment]
> Hello @YimiAChack ,
> 
> I have a question regarding the reproducibility of your code. The paper does not provide all the best hyperparameters for each dataset. Using the settings that I could infer, I ran the PeMSD4 experiment and obtained the following results, which are significantly different from those reported in the paper:
> 
> * RMSE: 32.63
> * MAE: 22.42
> * MAPE: 16.33
> 
> Below is the [log](https://github.com/user-attachments/files/18238194/log_2024-12-24_10-54-45_514918.log) from my experiment. I would greatly appreciate it if you could address this issue promptly.
> 
> the log from my exp;
> Thank you
May I ask if your issue has been resolved? The results of my experiment are roughly like this, which is very strange

**Comment by jeongwhanchoi** [comment]
> May I ask if your issue has been resolved? The results of my experiment are roughly like this, which is very strange

Hi @sxgty.
No, I still have a problem reproducing the results. We need to know the optimal hyperparameter settings.

**Comment by sxgty** [comment]
> > May I ask if your issue has been resolved? The results of my experiment are roughly like this, which is very strange
> 
> Hi @sxgty. No, I still have a problem reproducing the results. We need to know the optimal hyperparameter settings.

I am using the following version: Pytho3.8，torch2.1，numpy1.21，then running run_madel.py directly. But it did not achieve the performance stated in the paper. And may I ask one more thing? Have you noticed the STPGNN model accepted by AAAI2024 for traffic flow prediction? I am also unable to reproduce the results of this paper.

**Comment by sxgty** [comment]
I am using the following version: Pytho3.8，torch2.1，numpy1.21，then running run_madel.py directly. But it did not achieve the performance stated in the paper. And may I ask one more thing? Have you noticed the STPGNN model accepted by AAAI2024 for traffic flow prediction? I am also unable to reproduce the results of this paper.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jeongwhan ***@***.***&gt;; 
发送时间: 2025年1月1日(星期三) 下午4:37
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [YimiAChack/STDDE] Reproducibility Issue (Issue #2)














  
May I ask if your issue has been resolved? The results of my experiment are roughly like this, which is very strange
  
Hi @sxgty.
 No, I still have a problem reproducing the results. We need to know the optimal hyperparameter settings.

**Comment by YimiAChack** [comment]
Thank you for your attention! Unfortunately, due to the change in my affiliation, optimal parameters were not saved.  I recommend performing a grid search on the hyperparameters. Based on our experience, multi-step training is more costly, and you can try the one-step version first.

**Comment by YimiAChack** [comment]
> I am using the following version: Pytho3.8，torch2.1，numpy1.21，then running run_madel.py directly. But it did not achieve the performance stated in the paper. And may I ask one more thing? Have you noticed the STPGNN model accepted by AAAI2024 for traffic flow prediction? I am also unable to reproduce the results of this paper.

I have not run the STPGNN code in AAAI2024.

**Comment by jeongwhanchoi** [comment]
> Thank you for your attention! Unfortunately, due to the change in my affiliation, optimal parameters were not saved. I recommend performing a grid search on the hyperparameters. Based on our experience, multi-step training is more costly, and you can try the one-step version first.

Hi @YimiAChack , I understand the affiliation change caused some loss of experimental records. Would it be possible for you or any of the co-authors who contributed to the paper to recall and share the best hyperparameters used during the experiments? An approximate configuration in the README or directly in this issue thread would be greatly appreciated.
